test/socket.test.js

```diff
@@ -0,0 +1,35 @@
+const { app } = require('../src/expressServer');
+const io = require('socket.io-client');
+
+let server;
+const PORT = 8001;
+
+beforeAll(async () => {
+  server = await app.listen(PORT);
+});
+
+afterAll(async () => {
+  server.close();
+});
+
+test('test:echo socket event', (done) => {
+  jest.setTimeout(30000);
+  const client = io.connect(`http://localhost:${PORT}`, {
+    transports: ['websocket'],
+    forceNew: true,
+    reconnection: false,
+  });
+
+  const messageToSend = 'Hello, server!';
+
+  client.once('connect', () => {
+    client.emit('test:echo', messageToSend);
+
+    client.on('test:echo', (response) => {
```

Comment:
`test:echo` was for the example. But in this application we don't have a `test:echo` channel/event handler defined. We do have a `test:ping` channel though: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/PS192-socket-unit-test/src/live/index.js#L6](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/PS192-socket-unit-test/src/live/index.js#L6)

If we want to create a `test:echo` handler for the sake of this unit test exercise that's totally fine. Otherwise, we should be testing the existing `test:ping` handler.

Currently this test is failing because line 28 `client.on('test:echo')` is waiting and waiting for our server to send a websocket message to the `test:echo` listener, but there's no code in our backend doing that; so the test just hangs until Jest eventually kills it.